core/osrd-reporting/src/main/java/fr/sncf/osrd/reporting/exceptions/ErrorType.java

```diff
@@ -0,0 +1,354 @@
+            ErrorCause.USER
+    ),
+    UnknownAllowanceValueType(
+            "unknown_allowance_value_type",
+            "unknown allowance value type",
+            ErrorCause.USER
+    ),
+    MissingAttributeError(
+            "missing_attribute",
+            "referencing missing attribute",
+            ErrorCause.INTERNAL
+    ),
+    EnvelopeStopIndexOutOfBounds(
+            "envelope_error",
+            "Stop at index %d is out of bounds",
+            ErrorCause.USER
+    ),
+    EnvelopePartsNotContiguous(
+            "envelope_error",
+            "invalid envelope, envelope parts are not contiguous",
+            ErrorCause.USER
+    ),
+    NoCompatibleEnvelopeFound(
+            "no_compatible_envelope",
+            "Couldn't find an envelope that wouldn't cause a conflict",
+            ErrorCause.USER
+    ),
+    SignalingError(
+            "signaling_error",
+            "Unknown signaling system: $sigSystem",
+            ErrorCause.USER
+    ),
+    SigSchemaInvalidFieldError(
+            "sig_schema_field_error",
+            "invalid sig schema field",
+            ErrorCause.USER
+    ),
+    SigSchemaUnknownFieldError(
+            "sig_schema_field_error",
+            "unknown sig schema field",
+            ErrorCause.USER
+    ),
+
+    InvalidSTDCMDelayError(
+            "invalid_stdcm",
+            "STDCM lookahead isn't supported yet",
+            ErrorCause.USER
+    ),
+    InvalidSTDCMUnspecifiedStartTime(
+            "invalid_stdcm",
+            "STDCM requests with unspecified start time are not supported yet",
+            ErrorCause.USER
+    ),
+    InvalidSTDCMUnspecifiedStartAndEndTime(
+            "invalid_stdcm",
+            "Invalid STDCM request: both end time and start time are unspecified, at least one must be set",
+            ErrorCause.USER
+    ),
+    BALUnprotectedZones(
+            "unprotected_zones",
+            "BAL signals always protect zones",
+            ErrorCause.USER
+    ),
+    BAPRUnprotectedZones(
+            "unprotected_zones",
+            "BAPR signals always protect zones",
+            ErrorCause.USER
+    ),
+    UnexpectedReservationStatus(
+            "unexpected_reservation_status",
+            "unexpected reservation status",
+            ErrorCause.USER
+    ),
+    IncompatibleZoneRequirements(
+            "incompatible_zone_requirements",
+            "incompatible zone requirements",
+            ErrorCause.USER
+    ),
+    ActionLockRequired(
+            "action_lock_required",
+            "action lock required",
+            ErrorCause.USER
+    ),
+    BlockedSimulationException(
+            "blocked_simulation_exception",
+            "the discrete simulation blocked without completing",
+            ErrorCause.USER
+    )
+    ;
+
+    public final String type;
+    public final String message;
+    public final ErrorCause cause;
+
+    private ErrorType(String type, String message, ErrorCause cause) {
```

Comment:
It's implied for enum constructors

```suggestion
    ErrorType(String type, String message, ErrorCause cause) {
```